src/text_classification_pipeline.py

```diff
@@ -90,17 +93,17 @@ def __init__(
         self._embedder = self.str2enum(embeddings)
         self._checkpoint_path = self.get_checkpoint(algorithm, embeddings, class_balancer, preprocessor)
         self._mode = self.str2enum(mode)
+        self._preprocessed_data = None
+        self._vectorized_data = None
+        self._metrics = {}
     
     def run(self) -> None:
-        preprocess_data = self.preprocess()
-        vectorized_data = self.vectorize(preprocess_data)
+        self.preprocess()
 
         if self._mode == Mode.INFER:
-            self.infer(vectorized_data)
+            self.infer()
```

Comment:
self.infer() expect argument X, but we do not passing it. I think it is better to remove the `infer` mode from enum, because it requires own run command, command using now not appropriate to `infer` mode. So lets remove `infer` mode, but leaves the self.infer() method which is required into `eval` mode. As a result we will have two available mode: `train` and `eval`